src/wurf/dependency.py

```diff
@@ -79,34 +91,52 @@ def __init__(self, **kwargs):
         #    in external formats such as files
         # 2. Use the Python hash() function to generate a integer hash for 
         #    run-time usage (i.e. within the same interpreter session).
-        self.hash = None
+        #super(Dependency, self).__setattr__('hash', None)
+        self.read_only['hash'] = None
+        
+        # Resolver attributes (modifiable)
+        object.__setattr__(self, 'read_write', dict())
+        
         
     def __getattr__(self, option):
         """ Return the value corresponding to the option.
 
         :param option: The name of the option to return as a string.
         :return: The option value.
         """
+        if option in self.read_only:
+            return self.read_only[option]
+
+        elif option in self.read_write:
+            return self.read_write[option]
+        
+        else:
+            return None
+
+
+    def __setattr__(self, name, value): 
+        """ Sets a resolver attribute."""
 
-        if option in self.info:
-            return self.info[option]
+        if name in self.read_only:
+            raise AttributeError("Attribute {} read-only.".format(name))
         else:
-            raise AttributeError("No such attribute: " + option)
+            self.read_write[name] = value   
 
     def __contains__(self, option):
         """ Checks if the option is available.
         
         :return: True if the option is available otherwise False
         """
-        return option in self.info
+        return (option in self.read_only) or (option in self.read_write)
 
     def __str__(self):
-        return str(self.info)
+        return "Dependency\ninfo: {}\nresolver info: {}".format(
+            self.read_only, self.read_write)
 
     def __hash__(self):
         """ :return: Integer hash value for the dependency. """
         
-        if not self.hash:
-            self.hash = hash(self.info['sha1'])
+        if not self.read_only['hash']:
+            self.read_only['hash'] = hash(self.read_only['sha1'])
```

Comment:
The hash is also used by the registry caching functionality. 

In principle we can freely decide how what information should be used when calculating the hash. However, I would rather be very strict now.. see if it becomes a problem and the fix it. I worry that if it is not strict small difficult bugs could creep in where you do not see a mismatch between two dependencies causing a problem.. However, disregarding for example the `sources` attribute could potentially be safe.